core/ops.rs

```diff
@@ -20,19 +22,84 @@ use std::ops::DerefMut;
 use std::pin::Pin;
 use std::rc::Rc;
 
-pub type OpAsyncFuture = Pin<Box<dyn Future<Output = Box<[u8]>>>>;
-pub type OpFn = dyn Fn(Rc<RefCell<OpState>>, BufVec) -> Op + 'static;
+pub use erased_serde::Serialize as Serializable;
+pub type RcOpState = Rc<RefCell<OpState>>;
+pub type PromiseId = u64;
+pub type OpBuf = Option<ZeroCopyBuf>;
+pub type OpAsyncFuture = Pin<Box<dyn Future<Output = OpResponse>>>;
+pub type OpFn = dyn Fn(RcOpState, OpPayload, OpBuf) -> Op + 'static;
 pub type OpId = usize;
 
+pub struct OpPayload<'a, 'b, 'c> {
+  pub(crate) scope: Option<&'a mut v8::HandleScope<'b>>,
+  pub(crate) value: Option<v8::Local<'c, v8::Value>>,
+}
+
+impl<'a, 'b, 'c> OpPayload<'a, 'b, 'c> {
+  pub fn new(
+    scope: &'a mut v8::HandleScope<'b>,
+    value: v8::Local<'c, v8::Value>,
+  ) -> Self {
+    Self {
+      scope: Some(scope),
+      value: Some(value),
+    }
+  }
+
+  pub fn empty() -> Self {
+    Self {
+      scope: None,
+      value: None,
+    }
+  }
+
+  pub fn deserialize<T: DeserializeOwned>(self) -> Result<T, AnyError> {
+    // TODO: Map errors
+    serde_v8::from_v8(self.scope.unwrap(), self.value.unwrap())
+      .map_err(AnyError::from)
+  }
+}
+
+pub enum OpResponse {
+  Value(Box<dyn Serializable>),
+  Buffer(Box<[u8]>),
+}
```

Comment:
Just to make sure, this is temporary measure to provide compatiblity with json ops?

Reply:
This exists to allow returning buffers. This will likely evolve in follow-up PRs and is dependant on how we decide to implement @inteon's structured-serialization worker message-passing PR (which would introduce the only ops returning buffers)